src/sqlfluff/core/parser/grammar/anyof.py

```diff
@@ -197,7 +205,26 @@ def match(
             else:
                 pre_seg = ()  # empty tuple
 
-            match = self._match_once(unmatched_segments, parse_context=parse_context)
+            match, matched_option = self._match_once(
+                unmatched_segments, parse_context=parse_context
+            )
+
+            # Increment counter for matched option.
+            if (
+                available_option_counter
+                and matched_option
+                and (str(matched_option) in available_option_counter)
+            ):
+                available_option_counter[str(matched_option)] += 1
+            # Check if we have matched an option too many times.
```

Comment:
Could this be moved inside the `if` above? It may be possible to simplify the logic as well -- IIUC, we only need to check if `matched_option` has hit the maximum.

Reply:
@barrywhart good shout, I've simplified the logic by moving the if condition 👍